selfservice/hook/web_hook.go

```diff
@@ -352,6 +353,15 @@ func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
 
 		resp, err := httpClient.Do(req)
 		if err != nil {
+			if te := interface{ Timeout() bool }(nil); errors.As(err, &te) && te.Timeout() || errors.Is(err, context.DeadlineExceeded) {
```

Comment:
Getting to know if a HTTP request timed out is surprisingly hard in Go. That's an artifact from the days before `context`. If you use the deprecated `Timeout` field in the `http.Client, you get a `Timeout` error. If the request's context times out, you get `Context.DeadlineExceeded`.